jetbrains-core/tst/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/QueryingLogGroupsTest.kt

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudwatch.logs.insights
+
+import com.intellij.openapi.application.runWriteAction
+import com.intellij.openapi.projectRoots.ProjectJdkTable
+import com.intellij.openapi.roots.ProjectRootManager
+import com.intellij.testFramework.IdeaTestUtil
+import com.intellij.testFramework.RunsInEdt
+import com.intellij.testFramework.runInEdtAndWait
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import software.aws.toolkits.jetbrains.utils.rules.JavaCodeInsightTestFixtureRule
+import org.assertj.core.api.Assertions.assertThat
+import software.aws.toolkits.resources.message
+import java.util.Calendar
+import java.util.Date
+
+@RunsInEdt
+class QueryingLogGroupsTest {
+    @JvmField
+    @Rule
+    val projectRule = JavaCodeInsightTestFixtureRule()
+    private val qEditorValidator = QueryEditorValidator
+    private lateinit var view: QueryEditor
+
+    @Before
+    fun setup() {
+        val project = projectRule.project
+        val sdk = IdeaTestUtil.getMockJdk18()
+        runInEdtAndWait {
+            runWriteAction {
+                ProjectJdkTable.getInstance().addJdk(sdk, projectRule.fixture.projectDisposable)
+                ProjectRootManager.getInstance(projectRule.project).projectSdk = sdk
+            }
+            view = QueryEditor(project)
+        }
+    }
+
+    @Test
+    fun `Absolute or relative time selected`() {
+        getViewDetails(absoluteTime = false, relativeTime = false)
+        assertThat(qEditorValidator.validateEditorEntries(view)?.message).contains(message("cloudwatch.logs.validation.timerange"))
+    }
+
+    @Test
+    fun `Start date must be before end date`() {
+        val cal = Calendar.getInstance()
+        cal.add(Calendar.DATE, -1)
+        getViewDetails(absoluteTime = true, startDate = Calendar.getInstance().time, endDate = cal.time)
+        assertThat(qEditorValidator.validateEditorEntries(view)?.message).contains("Start date must be before end date")
+    }
+
+    @Test
+    fun `Number of time units not selected`() {
```

Comment:
nit: can you reword this to something along the lines of 'Relative time, no time entered'

Reply:
Done